From 7769d116768052de9a8f5f13abfefa4fc3fcf759 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20Koutn=C3=BD?= <mkoutny@suse.com>
Date: Fri, 22 Jun 2018 16:19:52 +0200
Subject: [PATCH 10/25] core/manager: Make sure jobs statistics are not double
 after daemon-reload

We add n_installed_jobs and n_failed_jobs to our inner state after
deserialization. This is fine during daemon-reexec when we start with clear
Manager (and some jobs possibly queued before deserialization), however,
daemon-reload works with the same manager and adding the values would
effectively double the counters. Reset the counters before we deserialize and
add their values again.

(cherry picked from commit 1e75824cb005f7fa0089792e45c2747c4d059601)
---
 src/core/manager.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/core/manager.c b/src/core/manager.c
index 930df4e23a..a3ae481158 100644
--- a/src/core/manager.c
+++ b/src/core/manager.c
@@ -1234,6 +1234,8 @@ static void manager_clear_jobs_and_units(Manager *m) {
 
         m->n_on_console = 0;
         m->n_running_jobs = 0;
+        m->n_installed_jobs = 0;
+        m->n_failed_jobs = 0;
 }
 
 Manager* manager_free(Manager *m) {
-- 
2.19.1

